csharp/ql/src/utils/model-generator/ModelGeneratorUtilsSpecific.qll

```diff
@@ -69,9 +70,14 @@ private predicate isPrimitiveTypeUsedForBulkData(Type t) {
   t.getName().regexpMatch("byte|char|Byte|Char")
 }
 
+private predicate isContainerType(Type t) {
```

Comment:
The old `LibraryTypeDataFlow.qll` used to have
```ql
private import semmle.code.csharp.frameworks.system.Collections

private predicate isCollectionType(ValueOrRefType t) {
  t.getABaseType*() instanceof SystemCollectionsIEnumerableInterface and
  not t instanceof StringType
}
```
which should cover all sorts of collection types (including arrays).

Reply:
Hah, that is interesting. I actually tried a solution similar to this one, but dropped it because a string would identify as a collection like type and I thought it would be a dit dirty to use an "exception" list (ie. t is a collection type if ..... unless t is a string, ...). I will re-introduce such a solution again :-)